### ISSUE TITLE
Support multiple values of the same key in defaultDirectives

### ISSUE THREAD

**Issue by qelix** [JonasKruckenberg/imagetools#139]
Hey @JonasKruckenberg,

thanks for the great project! Probably like most people, I want to use this plugin to create a bunch of differently sized images that are later served as an image with a `srcset` tag. Because I don't want to specify all of the breakpoints individually for each image, I found out that there's the `defaultDirectives` setting in the Vite plugin options. So the idea is to store a variable containing all of the breakpoints, which is then picked up by the Vite config and automatically applied to every imported image. 

However, the setting doesn't seem to support more than one value for each property at a time -- I tried passing the string exactly like you would in an import statement (`width=3840;2560;1920` etc.), but the plugin only seems to accept the first number it gets. The exact config I used looks like this:

```js
plugins: [
  imagetools({
    defaultDirectives: { width: imageWidths.join(";"), format: 'webp' },
  }),
],
```

Is there any way to solve this? I also thought about using dynamic import statements, but I never really got them to work properly. I really like your project, but the design decision to make everything a directive you have to spell out for each individual import makes it quite hard to outsource a common configuration. 

Your help would be appreciated :)

**Comment by JonasKruckenberg** [comment]
Sooo, I thought this was gonna come up at some point ;) 
The way imagetools works is it first parses the url and generates an object from that. It then generates what I call targets, one for each image that is gonna be produced.
The issue is that in your case the defaultDirectives get applied **after** the last step above. That's why it's not working.
I am however very unhappy with the way the API works right now so I'm doing (yet another *sigh*) rewrite and I'll make sure to address this issue too!

**Comment by JonasKruckenberg** [comment]
Alright let's think about this again, back in June I promised an API redesign that sadly went nowhere (for different reason but mostly time 😕) so lets move this forward and think of a way we can approach this without a redesign.
My first idea was to change `defaultDorectives` from
```ts
type defaultDirectives = Record<string, string> | ((id: string) => Record<string, string>)
```
to this
```ts
type defaultDirectives =  URLSearchParams | ((id: string) =>  URLSearchParams)
```
and merge them with the  `URLSearchParams` from the actual Url. This would resolve #216 since configs would be applied before resolving and allow for multiple values for the same key.
wdyt @Porges @qelix @jugglingcats would that solve you use case?

**Comment by Porges** [comment]
My use-case is I want to set one config for every single image to resize it to a set of standard sizes, which I then use in `srcset` to pick a different one depending on window size. I think what you’re describing will solve it.

(Another ability I need for this is to only resize downwards and never up-scale, but that’s another topic.)

**Comment by jugglingcats** [comment]
@JonasKruckenberg, I think default directives are nice and having them able to generate multiple images would be a common requirement. Our requirement is to be able to have multiple _custom presets_ that might generate a single image or multiple. The pull request I made achieves this _reasonably_ cleanly 🤞.

One change you might (also) consider is to default to returning metadata rather than just url when there are multiple images being generated. I can't really see a scenario where you'd generate multiple images but not need the metadata? At the moment I am using the following as well as our custom `resolveConfigs`:

```
export function glowsiteOutputFormats(builtinOutputFormats) {
    // all this does is ensure that when we write image.png?glowsite we get the metadata back not the urls
    // (it means we don't have to write image.png?glowsite&meta all the time)
    //
    // we also filter out a lot of the redundant metadata from the json as it ends up in the bundle
    return {
        ...builtinOutputFormats, glowsite: () => {
            const defaultFormat = metadataFormat()
            return (metadatas) => {
                // just return the metadata we need
                return defaultFormat(metadatas).map(({format, width, height, src}) => ({format, width, height, src}))
            }
        }
    }
}
```

**Comment by JonasKruckenberg** [comment]
Alright, the custom resolveConfig hook got released in `rollup-plugin-imagetools@2.5.0` and `vite-imagetools@3.9.0` respectively.
The direct fix for this issue will also be released soon!
